File: web/projects/portal/src/app/common/dnd/vs-table-dnd-event.ts
/*
 * inetsoft-web - StyleBI is a business intelligence web application.
 * Copyright © 2024 InetSoft Technology (info@inetsoft.com)
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
import { VSColumnDndEvent } from "./vs-column-dnd-event";
import { TableAddColumnInfo } from "./table-add-column-info";
import { TableRemoveColumnInfo } from "./table-remove-column-info";

/**
 * Event for dnd.
 */
export class VSTableDndEvent extends VSColumnDndEvent {
   /**
    * Creates a new instance of <tt>VSDndEvent</tt>.
    */
   constructor(name: string, addInfo: TableAddColumnInfo,
      removeInfo: TableRemoveColumnInfo, confirmed: boolean = true,
      checkTrap: boolean = false, sourceChanged: boolean = false,
      table: string = null)
   {
      super(name, confirmed, checkTrap, sourceChanged, table);
      this.addInfo = addInfo;
      this.removeInfo = removeInfo;
   }

   private addInfo: TableAddColumnInfo;
   private removeInfo: TableRemoveColumnInfo;
}

File: core/src/main/java/inetsoft/web/binding/event/VSTableDndEvent.java
/*
 * This file is part of StyleBI.
 * Copyright (C) 2024  InetSoft Technology
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */
package inetsoft.web.binding.event;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import inetsoft.web.binding.dnd.TableAddColumnInfo;
import inetsoft.web.binding.dnd.TableRemoveColumnInfo;
import org.immutables.value.Value;

import javax.annotation.Nullable;

@Value.Immutable
@JsonSerialize(as = ImmutableVSChartDndEvent.class)
@JsonDeserialize(as = ImmutableVSChartDndEvent.class)
public interface VSTableDndEvent extends VSColumnDndEvent {
   /**
    * Get the dnd transfer.
    * @return dnd transfer.
    */
   @Nullable
   TableAddColumnInfo addInfo();
   /**
    * Get the drop target.
    * @return drop target.
    */
   @Nullable
   TableRemoveColumnInfo removeInfo();
}
